list_test.go

```diff
@@ -66,21 +65,4 @@ var _ = Describe("List", func() {
 			Expect(quick.Check(f, nil)).To(Succeed())
 		})
 	})
-
-	Context("when constructing an empty list", func() {
-		It("should return an error", func() {
-			_, err := pack.NewList()
-			Expect(err).To(HaveOccurred())
-		})
-	})
-
-	Context("when unmarshaling an empty list", func() {
-		It("should return an error", func() {
-			list := pack.List{}
-			bytes, err := surge.ToBinary(list)
-			Expect(err).ToNot(HaveOccurred())
-			err = surge.FromBinary(&list, bytes)
-			Expect(err).To(HaveOccurred())
-		})
-	})
```

Comment:
We should keep these test cases, but explicitly check that they do *not* result in errors, but do result in Lists with the Nil element type.